store/snap_remote_repo.go

```diff
@@ -232,6 +255,123 @@ func (s *SnapUbuntuStoreRepository) checkStoreResponse(resp *http.Response) {
+
+HTTP/1.1 200 OK
+Content-Type: application/json; charset=utf-8
+
+[
+  {
+    "open_id": "https://login.staging.ubuntu.com/+id/open_id",
+    "package_name": "com.ubuntu.developer.dev.appname",
+    "refundable_until": "2015-07-15 18:46:21",
+    "state": "Complete"
+  },
+  {
+    "open_id": "https://login.staging.ubuntu.com/+id/open_id",
+    "package_name": "com.ubuntu.developer.dev.appname",
+    "item_sku": "item-1-sku",
+    "purchase_id": "1",
+    "refundable_until": null,
+    "state": "Complete"
+  },
+  {
+    "open_id": "https://login.staging.ubuntu.com/+id/open_id",
+    "package_name": "com.ubuntu.developer.dev.otherapp",
+    "refundable_until": "2015-07-17 11:33:29",
+    "state": "Complete"
+  }
+]
+*/
+type purchase struct {
+	OpenID          string `json:"open_id"`
+	PackageName     string `json:"package_name"`
+	RefundableUntil string `json:"refundable_until"`
+	State           string `json:"state"`
+	ItemSKU         string `json:"item_sku,omitempty"`
+	PurchaseID      string `json:"purchase_id,omitempty"`
+	RedirectTo      string `json:"redirect_to,omitempty"`
+}
+
+type purchaseList []purchase
+
+func (s *SnapUbuntuStoreRepository) getPurchasesFromURL(url *url.URL, auther Authenticator) (purchaseList, error) {
+	if auther == nil {
+		return nil, fmt.Errorf("cannot obtain known purchases from store: no authentication credentials provided")
+	}
+
+	req, err := http.NewRequest("GET", url.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	s.applyUbuntuStoreHeaders(req, "application/json", auther)
+
+	resp, err := s.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var purchases purchaseList
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		dec := json.NewDecoder(resp.Body)
+		if err := dec.Decode(&purchases); err != nil {
+			return nil, fmt.Errorf("cannot decode known purchases from store: %v", err)
+		}
+	case http.StatusNotFound:
+		// the store returns 404 for snaps that have no purchases, so just return an empty list
+		break
+	case http.StatusUnauthorized:
+		return nil, decodeUnauthorizedError(resp.Body)
+	default:
+		return nil, fmt.Errorf("cannot obtain known purchases from store: server returned %v code", resp.StatusCode)
+	}
+
+	return purchases, nil
+}
+
+// getPurchases retreives the user's purchases for a specific package, including in-app purchases, as a list
+func (s *SnapUbuntuStoreRepository) getPurchases(name string, auther Authenticator) (purchaseList, error) {
+	purchasesURL, err := s.purchasesURI.Parse(name + "/")
+	if err != nil {
+		return nil, err
+	}
+
+	q := purchasesURL.Query()
+	q.Set("include_item_purchases", "true")
+	purchasesURL.RawQuery = q.Encode()
+
+	return s.getPurchasesFromURL(purchasesURL, auther)
+}
+
+// getAllPurchases retreives all the user's purchases, including in-app purchases, as a map indexed by package name
+func (s *SnapUbuntuStoreRepository) getAllPurchases(auther Authenticator) (map[string]purchaseList, error) {
```

Comment:
this are all private? is that meant to be, and later there will be higher level, exported methods?